synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp

```diff
@@ -0,0 +1,257 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file valuenode_modulo.cpp
+**	\brief Implementation of the "Modulo" valuenode conversion.
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**	Copyright (c) 2022 Synfig Contributors
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+/* === H E A D E R S ======================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include "valuenode_modulo.h"
+#include "valuenode_const.h"
+
+#include <synfig/general.h>
+#include <synfig/localization.h>
+#include <synfig/valuenode_registry.h>
+#include <synfig/vector.h>
+#include <synfig/angle.h>
+#include <synfig/real.h>
+
+#include <ETL/misc>
+#include <ETL/stringf>
+
+#include <stdexcept>
+
+#endif
+
+/* === U S I N G =========================================================== */
+
+using namespace etl;
+using namespace synfig;
+
+/* === M A C R O S ========================================================= */
+
+/* === G L O B A L S ======================================================= */
+
+REGISTER_VALUENODE(ValueNode_Modulo, RELEASE_VERSION_1_4_0, "modulo", N_("Modulo"))
+
+/* === P R O C E D U R E S ================================================= */
+
+/* === M E T H O D S ======================================================= */
+
+ValueNode_Modulo::ValueNode_Modulo(const ValueBase &value):
+	LinkableValueNode(value.get_type())
+{
+	Vocab ret(get_children_vocab());
+	set_children_vocab(ret);
+	set_link("divisor",ValueNode_Const::create(int(2)));
+	set_link("scalar",ValueNode_Const::create(Real(1.0)));
+	Type& type(value.get_type());
+
+	if (type == type_angle)
```

Comment:
Why don't do something like:
```
if (!check_type(type))
{
	assert(0);
	throw std::runtime_error(get_local_name()+_(":Bad type ")+type.description.local_name);
}
set_link("link",ValueNode_Const::create(value));
```

Reply:
Nice.
I just based my code in another valuenode implementation (I think it was `add`) and didn't think about it :)
I'll fix it.

Reply:
Done :)